unit-tests/src/test/scala/scala/scalanative/native/TimeSuite.scala

```diff
@@ -2,139 +2,299 @@ package scala.scalanative.native
 
 import scala.scalanative.native.time.{
   asctime,
+  daylight,
   difftime,
+  gmtime,
   localtime,
   mktime,
   strftime,
-  wcsftime,
-  gmtime,
+  strptime,
   time_t,
-  tm
+  timezone,
+  tm,
+  tzname,
+  tzset,
+  wcsftime
 }
+
 import scala.scalanative.native.timeOps.tmOps
 
+import scala.scalanative.native.stdlib.{getenv, setenv, unsetenv}
+import scala.scalanative.native.string.{memset, strcmp}
+
 object TimeSuite extends tests.Suite {
 
-  val now_ptr: Ptr[time_t] = stackalloc[time_t]
-  val now_time_t: time_t   = time.time(now_ptr)
+  val now_time_t: time_t = time.time(null)
+
+  val tzEnvVarName     = c"TZ"
+  var savedTZ: CString = _
+
+  /* TimeSuite must execute in a single threaded environment. The only
+   * sure way to meet this requirement is for the enclosing testing
+   * environment to guarantee it.
+   *
+   * The four functions in the suite (localtime & friends) are defined
+   * as returning pointers to static buffers and THREAD_UNSAFE!.  In
+   * addition, this code possibly/probably changes the environment variable
+   * TZ for the duration of the suite run. Any other thread running in
+   * the same process may pick up a bogus, to it, value for TZ.
+   *
+   * A runtime test is possible here (stat /proc/self/task. If it
+   * is successful and nlinks == 3, then process is single threaded, at
+   * moment of testing). This test however is a weak one, because without
+   * a guarantee from the testing environment, a new thread can be created
+   * at any time after the initial test.
+   */
+
+  test("save unknown current TZ, set to known Central European Time (CET)") {
```

Comment:
Rather than relying on the ordering of the tests in this file, I'd personally do something like
```scala
  def addTest[R](name: String)(tz: String)(f: => R) = test(name) {
    val savedTZ = getenv(tzEnvVarName)
    try {
      assert(setenv(tzEnvVarnName, tz, 1) == 0)
      f
    } finally {
      assert(setenv(tzEnvVarName, name, 1) == 0 
    }
  }
```
It still isn't thread safe since another suite may modify the environment, but it at least allows one to isolate each specific test in the suite.

Reply:
@eatkins 

Thank you for the helpful suggestion.  I used to be aware of the technique but
but somehow it was not accessible to my brain when I was writing that code. Your
comment refresh my all too volatile memory.